lib/jsonapi/view.ex

```diff
@@ -229,15 +229,21 @@ defmodule JSONAPI.View do
         "#{url_for(data, conn)}/relationships/#{rel_type}"
       end
 
-      def url_for_pagination(data, conn, pagination_attrs) do
-        pagination_attrs
-        |> Enum.reduce(%{}, fn {key, value}, acc ->
-          Map.put(acc, "page[#{key}]", value)
-        end)
+      def url_for_pagination(data, %{query_params: query_params} = conn, pagination_attrs) do
+        query_params
+        |> Map.drop([:__struct__])
```

Comment:
Why do we need this `Map.drop([:__struct__])`? Looking at the `Plug.Conn` documentation, the `query_params` are just a map, not a struct.

Reply:
Without it, this test fail
```
1) test serialize includes pagination links if page-based pagination is requested (JSONAPI.SerializerTest)
     test/jsonapi/serializer_test.exs:585
     ** (Protocol.UndefinedError) protocol Enumerable not implemented for %{:__struct__ => Plug.Conn.Unfetched, :aspect => :query_params, "page" => %{"page" => "1", "size" => "1"}}. This protocol is implemented for: GenEvent.Stream, Range, HashDict, Function, HashSet, MapSet, Date.Range, Stream, Map, IO.Stream, List, File.Stream
     code: Serializer.serialize(PaginatedPostView, data, conn, nil, total_pages: 3, total_items: 3)
     stacktrace:
       (elixir) /home/build/elixir/lib/elixir/lib/enum.ex:1: Enumerable.impl_for!/1
       (elixir) /home/build/elixir/lib/elixir/lib/enum.ex:141: Enumerable.reduce/3
       (elixir) lib/enum.ex:3015: Enum.flat_map/2
       test/jsonapi/serializer_test.exs:69: JSONAPI.SerializerTest.PaginatedPostView.url_for_pagination/3
       test/jsonapi/serializer_test.exs:45: JSONAPI.SerializerTest.PageBasedPaginator.paginate/5
       (jsonapi) lib/jsonapi/serializer.ex:179: JSONAPI.Serializer.merge_links/7
       test/jsonapi/serializer_test.exs:595: (test)
```
But I only have to call `Plug.Conn.fetch_query_params` on the `conn` before passing it through the test instead of the solution above.